napari/_qt/qt_dims.py

```diff
@@ -218,41 +234,57 @@ def _update_nsliders(self):
             if self._displayed_sliders[i]:
                 self._update_slider(i)
 
-    def _create_sliders(self, number_of_sliders):
-        """
-        Creates sliders to match new number of dimensions
+    def _update_axis_labels(self, axis):
+        """Updates the label for the given axis """
+        self.axis_labels[axis].setText(self.dims.axis_labels[axis])
+
+    def _create_sliders(self, number_of_sliders: int):
+        """Creates sliders to match new number of dimensions
 
         Parameters
         ----------
-        number_of_sliders : new number of sliders
+        number_of_sliders : int
+            new number of sliders
         """
         # add extra sliders so that number_of_sliders are present
         # add to the beginning of the list
         for slider_num in range(self.nsliders, number_of_sliders):
             dim_axis = number_of_sliders - slider_num - 1
+            axis_label = self._create_axis_label_widget(dim_axis)
             slider = self._create_range_slider_widget(dim_axis)
-            self.layout().addWidget(slider)
+
+            # Hard-coded 1:50 ratio. Can be more dynamic as a function
+            # of the name of the label, but it might be a little bit
+            # over the top.
+            current_row = QHBoxLayout()
+            if axis_label.text != '':
+                current_row.addWidget(axis_label, stretch=1)
+                current_row.addWidget(slider, stretch=50)
+            else:
+                current_row.addWidget(slider)
+            self.layout().addLayout(current_row)
+            self.axis_labels.insert(0, axis_label)
             self.sliders.insert(0, slider)
             self._displayed_sliders.insert(0, True)
             nsliders = np.sum(self._displayed_sliders)
             self.setMinimumHeight(nsliders * self.SLIDERHEIGHT)
 
     def _trim_sliders(self, number_of_sliders):
-        """
-        Trims number of dimensions to a lower number
+        """Trims number of dimensions to a lower number
 
         Parameters
         ----------
-        number_of_sliders : new number of sliders
+        number_of_sliders : int
+            new number of sliders
         """
         # remove extra sliders so that only number_of_sliders are left
         # remove from the beginning of the list
         for slider_num in range(number_of_sliders, self.nsliders):
             self._remove_slider(0)
 
     def _remove_slider(self, index):
-        """
-        Remove slider at index
+        """Remove slider at index. Should also remove all accompanying widgets,
+        like the axis label.
```

Comment:
Still not compliant with PEP257: should be one line, with a longer description if needed after a blank line.